src/screens/ENSConfirmRegisterSheet.js

```diff
@@ -0,0 +1,214 @@
+import { RegistrationReviewRows } from '../components/ens-registration';
+import { GasSpeedButton } from '../components/gas';
+import { SheetActionButtonRow, SlackSheet } from '../components/sheet';
+import { RapActionTypes } from '../raps/common';
+import {
+  AccentColorProvider,
+  Box,
+  Divider,
+  Heading,
+  Inline,
+  Inset,
+  Stack,
+  Text,
+} from '@rainbow-me/design-system';
+import { ENS_DOMAIN, generateSalt } from '@rainbow-me/helpers/ens';
+import {
+  useAccountSettings,
+  useCurrentNonce,
+  useENSProfile,
+  useENSRegistration,
+  useENSRegistrationCosts,
+  useGas,
+} from '@rainbow-me/hooks';
+import { ImgixImage } from '@rainbow-me/images';
+import { loadWallet } from '@rainbow-me/model/wallet';
+import { getRapEstimationByType } from '@rainbow-me/raps';
+import { saveCommitRegistrationParameters } from '@rainbow-me/redux/ensRegistration';
+
+export const ENSConfirmRegisterSheetHeight = 600;
+const secsInYear = 31536000;
+const avatarSize = 70;
+
+export default function ENSConfirmRegisterSheet() {
+  const dispatch = useDispatch();
+  const { gasFeeParamsBySpeed, updateTxFee, startPollingGasFees } = useGas();
+  const { name: ensName, records } = useENSProfile();
+  const { accountAddress, network } = useAccountSettings();
+  const getNextNonce = useCurrentNonce(accountAddress, network);
+  const [gasLimit, setGasLimit] = useState();
+  const { params } = useRoute();
+
+  const [duration, setDuration] = useState(1);
+
+  const name = ensName.replace(ENS_DOMAIN, '');
+  const { data: registrationData } = useENSRegistration({
+    name,
+  });
+  const rentPrice = registrationData?.rentPrice;
+  const { data: registrationCostsData } = useENSRegistrationCosts({
+    duration,
+    name,
+    rentPrice,
+  });
+
+  const updateGasLimit = useCallback(async () => {
+    const salt = generateSalt();
+    const gasLimit = await getRapEstimationByType(RapActionTypes.commitENS, {
+      ensRegistrationParameters: {
+        duration: secsInYear,
+        name: name,
+        ownerAddress: accountAddress,
+        records,
+        rentPrice,
+        salt,
+      },
+    });
+    updateTxFee(gasLimit);
+    setGasLimit(gasLimit);
+  }, [accountAddress, name, records, rentPrice, updateTxFee]);
+
+  // Update gas limit
+  useEffect(() => {
+    if (!gasLimit && !isEmpty(gasFeeParamsBySpeed)) {
+      updateGasLimit();
+    }
+  }, [gasFeeParamsBySpeed, gasLimit, updateGasLimit, updateTxFee]);
+
+  useEffect(() => startPollingGasFees(), [startPollingGasFees]);
+
+  const handleCommitSubmit = useCallback(async () => {
+    const wallet = await loadWallet();
+    if (!wallet) {
+      return;
+    }
+
+    const nonce = await getNextNonce();
+    const salt = generateSalt();
+
+    const ensRegistrationParameters = {
+      duration: secsInYear,
+      name,
+      nonce,
+      ownerAddress: accountAddress,
+      records,
+      rentPrice,
+      salt,
+    };
+
+    await dispatch(
```

Comment:
do we really need to await here?

Reply:
https://github.com/rainbow-me/rainbow/pull/2975